Record decision about publishing libraries and version constraints (#4968)
input-output-hk/cardano-wallet: docs/site/src/contributor/decisions/2025-02-03-dependency-versions.md

## **Decision**

* We set version bounds constraints at the level of individual `.cabal` files instead of the `.project` file
* We update version bounds as needed, tracking upstream Cardano dependencies as they are released to _CHaP_
* We publish useful libraries to CHaP regularly, tracking upstream dependencies
* We monitor the use of those libraries

## **Details**

* All components of the cardano-wallet are currently versioned using `0.yyyy.mm.dd` pattern. In order to ensure published libraries are deemed newer, we need to change their versioning scheme to the usual `1.x.y.z` (eg. following [PVP](https://pvp.haskell.org))
* We need to start from the bottom-up with [cardano-wallet-read](https://github.com/cardano-foundation/cardano-wallet-agda/blob/f9709fafae2585f24f9833e71bf0b4de3023b3ed/lib/cardano-wallet-read/cardano-wallet-read.cabal#L1) package which lives in the cardano-wallet-agda repository

> * User-facing libraries might depend on "technical" libraries which are mostly useless and implementation details. We nevertheless need to publish those packages to CHaP in order for Cabal resolution mechanism to work properly

Reviewer: ```suggestion
* User-facing libraries might depend on "technical" libraries which mostly expose implementation details. We nevertheless need to publish those packages to CHaP in order for Cabal resolution mechanism to work properly
```

I do hope that they are useful — just not generally so. 😅

Author: Yeah, thanks for the rephrasing 🤦
